src/test/java/io/stargate/sgv2/jsonapi/service/operation/ReadAttemptTest.java

```diff
@@ -0,0 +1,68 @@
+package io.stargate.sgv2.jsonapi.service.operation;
+
+import static org.mockito.Mockito.mock;
+
+import com.datastax.oss.driver.api.core.metadata.Node;
+import com.datastax.oss.driver.api.core.servererrors.InvalidQueryException;
+import io.stargate.sgv2.jsonapi.exception.WarningException;
+import io.stargate.sgv2.jsonapi.fixtures.testdata.TestData;
+import java.util.List;
+import org.junit.jupiter.api.Test;
+
+/** Unit tests for {@link ReadAttempt}. */
+public class ReadAttemptTest {
+
+  private static final TestData TEST_DATA = new TestData();
+
+  @Test
+  public void retryOnAllowFilteringError() {
+
+    var msg = "retryOnAllowFilteringError()";
+    var mockNode = mock(Node.class);
+    var allowFilteringError =
+        new InvalidQueryException(
+            mockNode,
+            "Cannot execute this query as it might involve data filtering and thus may have unpredictable performance. If you want to execute this query despite the performance unpredictability, use ALLOW FILTERING");
+
+    var fixture = TEST_DATA.readAttempt().emptyReadAttemptFixture();
+
+    var cql1 =
+        "SELECT * FROM %s.%s"
+            .formatted(
+                TEST_DATA.names.KEYSPACE_NAME.asCql(true), TEST_DATA.names.TABLE_NAME.asCql(true));
+    var cql2 =
+        "SELECT * FROM %s.%s ALLOW FILTERING"
+            .formatted(
+                TEST_DATA.names.KEYSPACE_NAME.asCql(true), TEST_DATA.names.TABLE_NAME.asCql(true));
+
+    fixture
+        .attempt()
+        .setStatus(OperationAttempt.OperationStatus.READY)
+        .queryExecutor()
+        .doExecuteReadThrowThenReturn(allowFilteringError, fixture.resultSet())
+        .attempt()
+        .assertCompleted()
```

Comment:
Nit, wonder if the name should indicate it actually call execute for the attempt.
Similar to doExecuteReadThrowThenReturn() above, that one is more clear.

Reply:
discussing - assertExecuteCompleted() 

maybe also change the doExcecute etc to be when* - they were called do because it was originally using do() from mock library